centos/common/network-config.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+sed -i '/\[main\]/a no-auto-default=*' /etc/NetworkManager/NetworkManager.conf
```

Comment:
Thanks @mvrequa  for the PR. Please confirm that this is targeted only for eht0.

Reply:
no-auto-default=* changes NetworkManager from opt-out, to opt-in. As it, don't manage the network devices unless the config (from cloud-init) specifically calls for it.
Our partners in AzLinux added code in cloud-init which will configure only interfaces which have address in [IMDS](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service?tabs=linux)
https://github.com/canonical/cloud-init/pull/709

VLAN NICs will have an IMDS interface, IB NICs will not. So cloud init will define the 
This will work for all combinations of VLAN nics and IB nics and in any order since IMDS will have the VLAN but not the IB.
Testing included 2 VLAN nics and 1 IB nic configuration.

Reply:
See on 3/2 synthetic VM start goes from ~40s to ~4s (35s faster!).

Date | Service | ct | p50 | p90 | p99
-- | -- | -- | -- | -- | --
2021-02-28 00:00:00.0000000 | AzLinux | 227 | 40352 | 47947 | 51294
2021-03-01 00:00:00.0000000 | AzLinux | 231 | 38769 | 47068 | 49364
2021-03-02   00:00:00.0000000 | AzLinux | 256 | 2553 | 4482 | 5416
2021-03-03 00:00:00.0000000 | AzLinux | 158 | 2537 | 5009 | 14331
2021-03-04 00:00:00.0000000 | AzLinux | 83 | 4423 | 4606 | 13445
2021-03-05 00:00:00.0000000 | AzLinux | 68 | 4434 | 4585 | 4912
2021-03-06 00:00:00.0000000 | AzLinux | 85 | 4408 | 4656 | 10311
2021-03-07 00:00:00.0000000 | AzLinux | 91 | 4435 | 4671 | 7997
2021-03-08 00:00:00.0000000 | AzLinux | 65 | 4424 | 4603 | 4736

Reply:
Awesome, thanks!